pkg/chunked/compression_linux.go

```diff
@@ -157,10 +162,41 @@ func readEstargzChunkedManifest(blobStream ImageSourceSeekable, blobSize int64,
 	return manifestUncompressed, tocOffset, nil
 }
 
+func openTmpFile(tmpDir string) (fd int, err error) {
+	fd, err = unix.Open(tmpDir, unix.O_TMPFILE|unix.O_RDWR|unix.O_CLOEXEC|unix.O_EXCL, 0o600)
```

Comment:
minor, but should be able to use https://pkg.go.dev/os#OpenFile here and below to create an `*os.File` directly - can use the flags from https://pkg.go.dev/syscall similar to the re-exported constants in pkg/os (https://pkg.go.dev/os#pkg-constants)

Reply:
thanks, amended now